plugins/pulp_rpm/plugins/migrations/0031_yum_metadata_storage_path.py

```diff
@@ -0,0 +1,39 @@
+from gettext import gettext as _
+import logging
+import os
+import tempfile
+
+from pulp.server.db import model
+from pulp_rpm.plugins.db import models
+
+
+_log = logging.getLogger('pulp')
+
+
+def migrate(*args, **kwargs):
+    """
+    Re-calculate storage path for yum_repo_metadata_file unit
+    """
+    metadata_files = models.YumMetadataFile.objects.filter()
+    repos = set()
+    for unit in metadata_files:
```

Comment:
This is a small suggestion, but I find it useful especially for testability to create a separate function that migrates just one unit. Then you can separate the workflow of getting a connection, getting the db, querying for units, and looping from the act of modifying a single unit.

In this case, that function could return a repo_id perhaps, and let the outer function keep track of the accumulated repo IDs.